impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -367,25 +368,65 @@ void RenderPassVK::SetInstanceCount(size_t count) {
   instance_count_ = count;
 }
 
-// |RenderPass|
-bool RenderPassVK::SetVertexBuffer(VertexBuffer buffer) {
-  vertex_count_ = buffer.vertex_count;
-  if (buffer.index_type == IndexType::kUnknown || !buffer.vertex_buffer) {
+static bool AppendVertexBuffer(
+    CommandBufferVK& command_buffer,
+    std::vector<vk::Buffer>& vertex_buffers,
+    std::vector<vk::DeviceSize>& vertex_buffer_offsets,
+    const BufferView& vertex_buffer_view) {
+  if (!vertex_buffer_view) {
+    return false;
+  }
+  auto vertex_buffer = vertex_buffer_view.buffer;
+
+  if (!vertex_buffer) {
+    VALIDATION_LOG << "Failed to acquire device buffer"
+                   << " for vertex buffer view";
     return false;
   }
 
-  if (!command_buffer_->Track(buffer.vertex_buffer.buffer)) {
+  if (!command_buffer.Track(vertex_buffer)) {
     return false;
   }
 
-  // Bind the vertex buffer.
-  vk::Buffer vertex_buffer_handle =
-      DeviceBufferVK::Cast(*buffer.vertex_buffer.buffer).GetBuffer();
-  vk::Buffer vertex_buffers[] = {vertex_buffer_handle};
-  vk::DeviceSize vertex_buffer_offsets[] = {buffer.vertex_buffer.range.offset};
+  auto vertex_buffer_handle = DeviceBufferVK::Cast(*vertex_buffer).GetBuffer();
+  vertex_buffers.push_back(vertex_buffer_handle);
+  vertex_buffer_offsets.push_back(vertex_buffer_view.range.offset);
 
-  command_buffer_vk_.bindVertexBuffers(0u, 1u, vertex_buffers,
-                                       vertex_buffer_offsets);
+  return true;
+}
+
+// |RenderPass|
+bool RenderPassVK::SetVertexBuffer(VertexBuffer buffer) {
+  std::vector<vk::Buffer> vertex_buffers;
```

Comment:
See comment above.

With a fixed maximum vertex buffer size you can avoid heap allocation for binding vertex buffers.

Something like std::array of length 8 and an offset. Same for the metal pass.